File: node/check.js
const { DNSimple, AuthenticationError } = require('dnsimple');

(async () => {
  const dnsimple = new DNSimple({
    baseUrl: 'https://api.sandbox.dnsimple.com',
    accessToken: process.env.TOKEN,
    userAgent: 'dnsimple-examples',
  });

  try {
    const identity = await dnsimple.identity.whoami();
    const accountId = identity.data.account.id;
    const domainName = process.env.DOMAIN || 'dns.com';

    const check = await dnsimple.registrar.checkDomain(accountId, domainName);
    console.log(check);
  } catch (err) {
    if (err instanceof AuthenticationError) {
      console.error('Authentication error. Check your token is correct for the sandbox environment.');
    } else {
      console.error(err);
    }
  }
})();

File: elixir/check.exs
# Usage: mix run check.exs example.com

client = %Dnsimple.Client{access_token: Application.get_env(:dnsimple, :access_token), base_url: "https://api.sandbox.dnsimple.com/"}

{:ok, response} = Dnsimple.Identity.whoami(client)
account_id = response.data.account.id

name = Enum.at(System.argv, 0)

IO.inspect Dnsimple.Registrar.check_domain(client, account_id, name)
